### ISSUE TITLE
Inbound endpoint when coordination is false not working as expected in a worker manager cluster

### ISSUE THREAD

**Issue by chanikag** [wso2/product-ei#1857]
**Description:**
Tested with a JMS inbound endpoint in a worker -manager cluster (1 worker) in ESB 5.0.0. These are the findings,

1. In nonWorkerManager cluster when the coordination is false no subscription is added in the broker
2. In worker-manager cluster when the coordination is false no subscription is added
3. In worker-manager cluster when only the worker node is up when the coordination is false the subscription is added
4. If the coordination is true the subscription is added irrespective of the clustering pattern

In other words, when the coordination is false all the nodes should be started as worker nodes. Then multiple subscriptions were added. So we need to fix this to behave as expected (when coordination is false add multiple subscriptions for inbound endpoints) when clustering pattern is nonWorkerManager.

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

**Comment by nandika** [comment]
Can you verify this with EI 6.2.0 snapshot version also, With nonWorkerManager configuration

**Comment by Nirothipan** [comment]
This issue is not reproducible in EI 6.1.0 update 22 with coordination = "false".

Followed the following steps, 

COnfigured 2 non-worker manager nodes.

Created JMS inbound endpoint with co-ordination false.

------------------------------------------------------------------------------------

<?xml version="1.0" encoding="UTF-8"?>
<inboundEndpoint xmlns="http://ws.apache.org/ns/synapse"
                 name="testJMSClustering"
                 sequence="sendOut"
                 onError="onError"
                 protocol="jms"
                 suspend="false">
   <parameters>
      <parameter name="interval">1000</parameter>
      <parameter name="sequential">true</parameter>
      <parameter name="coordination">false</parameter>
      <parameter name="java.naming.factory.initial">org.wso2.andes.jndi.PropertiesFileInitialContextFactory</parameter>
      <parameter name="java.naming.provider.url">conf/jndi.properties</parameter>
      <parameter name="transport.jms.ConnectionFactoryJNDIName">QueueConnectionFactory</parameter>
      <parameter name="transport.jms.ConnectionFactoryType">queue</parameter>
      <parameter name="transport.jms.Destination">JMSMS</parameter>
      <parameter name="transport.jms.SessionAcknowledgement">AUTO_ACKNOWLEDGE</parameter>
      <parameter name="transport.jms.CacheLevel">3</parameter>
      <parameter name="transport.jms.SubscriptionDurable">false</parameter>
      <parameter name="transport.jms.ContentType">application/xml</parameter>
      <parameter name="transport.jms.SharedSubscription">false</parameter>
   </parameters>
</inboundEndpoint>

-----------------------------------------------------------------------

Behavior observed.

When both the nodes started, two subscriptions were added to MB.
When messages were sent to send to Inbound Queue, it was distributed among both the nodes.

--------------------------------------------------------------------------------

When coordination was set to true in the inbound endpoint configuration, only one subscription was added. and single node served the request and the request was switched to the other node when the node previously serving got down.

Hence this issue is not valid in EI -6.2.0 and can be closed.

**Comment by Nirothipan** [comment]
Similiar issue:- https://github.com/wso2/product-ei/issues/532

**Comment by Nirothipan** [comment]
Should be fixed with Similiar issue:- https://github.com/wso2/product-ei/issues/532

**Comment by Nirothipan** [comment]
Should have been fixed with https://github.com/wso2/product-ei/issues/532
